client/client.go

```diff
@@ -350,64 +417,32 @@ func newHttpsClient(conf Config) (*http.Client, error) {
 
 // Client configuration
 
-type Config struct {
-	ServerCert string
-	IsHttps    bool
-	NoVerify   bool
-}
-
-type systemCertPoolGetter interface {
-	GetSystemCertPool() (*x509.CertPool, error)
-}
-
-type systemCertPool struct{}
-
-func (systemCertPool) GetSystemCertPool() (*x509.CertPool, error) {
-	return x509.SystemCertPool()
-}
-
-func loadServerTrust(conf *Config) *x509.CertPool {
-	return loadServerTrustImpl(conf, systemCertPool{})
+// HttpsClient holds the configuration for the client side mTLS configuration
```

Comment:
It's not obvious from the context where this is used. I would consider adding a comment: "Careful when changing this. The struct is directly exposed in the mender.conf file."

Reply:
True! Maybe a naming change is better? 
`HttpsClientConfig`?

Reply:
For good style and self containment, I would argue that the struct should live in `conf/config.go`, and either we would use that struct from `client/client.go`, or it can use a different struct altogether.

Reply:
Yes, that's fine.